tools/raisinbread_refresh.php

```diff
@@ -0,0 +1,222 @@
+    $password = $dbInfo['password'];
+
+    $urlConfigSetting = $config->getSetting('url');
+    $baseConfigSetting = $config->getSetting('base');
+    $hostConfigSetting = $config->getSetting('host');
+
+} catch (\DatabaseException $e) {
+    echo 'Could not connect to the database in the Config file.' .
+        'It\'s possible that the LORIS tables have already been dropped.' .
+        PHP_EOL;
+    echo 'Continue attempting to install Raisinbread database? (yN)' . PHP_EOL;
+    $input = trim(fgets(STDIN));
+
+    if (mb_strtolower($input) !== 'y') {
+        die;
+    }
+    echo PHP_EOL . 'Please enter the name of your database:' . PHP_EOL;
+    $dbname = trim(fgets(STDIN));
+} catch (\LorisException $e) {
+    die($e->getMessage());
+} catch (Exception $e) {
+    die("Could not load project/config.xml");
+}
+
+
+echo <<<CONFIRMATION
+Please re-type the database name `$dbname` to confirm you wish to drop tables
+and import test data: 
+CONFIRMATION;
+
+$input = trim(fgets(STDIN));
+if ($input !== $dbname) {
+    die('Input did not match database name. Exiting.');
+}
+
+// Create a connection to mysql via bash. All credentials are supplied via 
+// command line arguments except for the password which must be entered manually.
+$mysqlCommand = "mysql -A \"$dbname\"";
+
+// Drop tables
+echo PHP_EOL .'Dropping LORIS tables....' . PHP_EOL;
+dropTables();
+
+
+// Source LORIS core tabes
+echo <<<INFO
+[*] Creating LORIS core tables.... 
+
+INFO;
+$coreTables = glob("../SQL/0000-*.sql"); 
+array_walk($coreTables, 'runPatch');
+
+// Create instrument tables
+echo <<<INFO
+[*] Creating instrument tables.... 
+
+INFO;
+$rbInstrumentTables = glob("../raisinbread/instruments/instrument_sql/*.sql"); 
+array_walk($rbInstrumentTables, 'runPatch');
+
+// Import Raisinbread data
+echo <<<INFO
+[*] Importing Raisinbread data....
+
+INFO;
+$rbData= glob("../raisinbread/RB_files/*.sql");
+array_walk($rbData, 'runPatch');
+
+// Restore config settings if they were successfully found before.
+if (isset($urlConfigSetting)) {
+    restoreConfigSetting('url', $urlConfigSetting);
+}
+if (isset($baseConfigSetting)) {
+    restoreConfigSetting('base', $baseConfigSetting);
+}
+if (isset($hostConfigSetting)) {
+    restoreConfigSetting('host', $hostConfigSetting);
+}
+
+// Trigger a password reset because the password for `admin` in the Raisinbread
+// database is public.
+echo 'Please choose a new password for the admin user:' . PHP_EOL;
+runCommand('php resetpassword.php admin');
+
+
+// END SCRIPT
+
+
+/**
+ * Drops core LORIS tables as well as Raisinbread instrument tables
+ */
+function dropTables(): void
+{
+    runPatch(
+        '../raisinbread/instruments/instrument_sql/9999-99-99-drop_instrument_tables.sql'
+        );
+    runPatch('../SQL/9999-99-99-drop_tables.sql');
+}
+
+function runPatch(string $file): void
+{
+    global $mysqlCommand;
+    runCommand(
+        "cat $file | $mysqlCommand"
+    );
+}
+
+function sqlFileFilter(string $file): bool
```

Comment:
what does this function do ?

Reply:
It's leftover from an earlier approach I think. I'll remove it.